src/__tests__/lib/containers/entity/annotations/SchemaDrivenAnnotationsEditor.test.tsx

```diff
@@ -0,0 +1,459 @@
+    renderComponent()
+    await screen.findByText('requires scientific annotations', { exact: false })
+    await screen.findByLabelText('country*')
+  })
+
+  it('Handles conditional schemas', async () => {
+    server.use(noAnnotationsHandler, ...schemaHandlers)
+    renderComponent()
+    await screen.findByText('requires scientific annotations', { exact: false })
+    const countryField = await screen.findByLabelText('country*')
+
+    // The required "state" field should not appear unless we select USA
+    expect(screen.queryByLabelText('state*')).not.toBeInTheDocument()
+
+    // Behavior under test: select "USA" and "state" field appears
+    userEvent.selectOptions(countryField, 'USA')
+    await screen.findByLabelText('state*')
+  })
+
+  it('Handles annotations with no schema', async () => {
+    server.use(annotationsHandler, noSchemaHandler)
+
+    renderComponent()
+
+    // custom fields are always arrays
+    // we need the '-0' suffix to reference the first textbox
+    const countryField = (await screen.findByRole('textbox', {
+      name: 'country-0',
+    })) as HTMLInputElement
+    expect(countryField.value).toBe('USA')
+    const stateField = (await screen.findByRole('textbox', {
+      name: 'state-0',
+    })) as HTMLInputElement
+    expect(stateField.value).toBe('Washington')
+
+    expect(
+      screen.queryByText('requires scientific annotations', {
+        exact: false,
+      }),
+    ).not.toBeInTheDocument()
+  })
+
+  it('Fetches existing annotations and schema and loads them into the form', async () => {
+    server.use(annotationsHandler, ...schemaHandlers)
+    renderComponent()
+    await screen.findByText('requires scientific annotations', { exact: false })
+    const countryField = (await screen.findByLabelText(
+      'country*',
+    )) as HTMLInputElement
+    const stateField = (await screen.findByLabelText(
+      'state*',
+    )) as HTMLInputElement
+    expect(countryField.value).toBe('USA')
+    expect(stateField.value).toBe('Washington')
+  })
+
+  it('Removes a custom annotation field when the last value is removed', async () => {
+    server.use(annotationsHandler, noSchemaHandler)
+    renderComponent()
+
+    // Should be able to find the 'country' custom field
+    await screen.findByRole('textbox', {
+      name: 'country-0',
+    })
+
+    // Remove the last element
+    userEvent.click(await screen.findByLabelText('Remove country-0'))
+
+    expect(
+      screen.queryByRole('textbox', {
+        name: 'country-0',
+      }),
+    ).not.toBeInTheDocument()
+  })
+
+  it('Sends a request to update annotations (no schema)', async () => {
+    server.use(annotationsHandler, noSchemaHandler, successfulUpdateHandler)
+
+    renderComponent()
+
+    await clickSave()
+
+    expect(updatedJsonCaptor).toBeCalledWith(
+      expect.objectContaining({ country: ['USA'], state: ['Washington'] }),
+    )
+  })
+
+  // TODO: This test is unstable, unclear why
```

Comment:
There are two unstable tests in this file, and both are set to be skipped. The tests always pass when run in isolation, but I can't figure out what aspect of each test isn't being isolated when run normally.

Reply:
To elaborate, the tests occasionally fail when checking that the `onSuccess` function was called (in `clickSave`)